src/_zkapauthorizer/model.py

```diff
@@ -93,20 +96,21 @@ class NotEnoughTokens(Exception):
 # all.  The schema inside the database is versioned by yet another mechanism.
 CONFIG_DB_NAME = "privatestorageio-zkapauthz-v1.sqlite3"
 
+_ConnectParamSpec = [str, int, bool, str, bool]
+_ConnectFunction = Callable[_ConnectParamSpec, Connection]
 
-def open_and_initialize(path, connect=None):
+
+def open_and_initialize(path: FilePath, connect: _ConnectFunction) -> Connection:
```

Comment:
`connect` is still optional at the higher level API `VoucherStore.from_node_config` but there seemed to be no reason to provide a default for this low-level API.